src/Microsoft.Identity.Web/WebApiExtensions/MicrosoftIdentityWebApiAuthenticationBuilderExtensions.cs

```diff
@@ -142,7 +142,6 @@ public static class MicrosoftIdentityWebApiAuthenticationBuilderExtensions
             bool subscribeToJwtBearerMiddlewareDiagnosticsEvents)
         {
             builder.AddJwtBearer(jwtBearerScheme, configureJwtBearerOptions);
-            builder.Services.Configure(jwtBearerScheme, configureMicrosoftIdentityOptions);
```

Comment:
I don't think this will work because we need the MicrosoftIdentityOptions as well, if they are used.

Reply:
For example: https://github.com/AzureAD/microsoft-identity-web/blob/master/tests/DevApps/WebAppCallsWebApiCallsGraph/Client/Startup.cs#L79 but on the web API side.

Reply:
Hi @jennyf19, could you help me understand how the MicrosoftIdentityOptions wouldn't be included?  I only see two
code paths and both make a Services.Configure call using the same JwtBearerOptions and MicrosoftIdentityOptions. Full paths I'm following below:

The private function [AddMicrosoftIdentityWebApiImplementation](https://github.com/AzureAD/microsoft-identity-web/blob/48e4a5e9af22439ac9e633707860073bd5d4c91e/src/Microsoft.Identity.Web/WebApiExtensions/MicrosoftIdentityWebApiAuthenticationBuilderExtensions.cs#L137) I deleted the builder.Services.Configure call from is called by the two implementations of the AddMicrosoftIdentityWebApi method in the same file ([1](https://github.com/AzureAD/microsoft-identity-web/blob/48e4a5e9af22439ac9e633707860073bd5d4c91e/src/Microsoft.Identity.Web/WebApiExtensions/MicrosoftIdentityWebApiAuthenticationBuilderExtensions.cs#L74), [2](https://github.com/AzureAD/microsoft-identity-web/blob/48e4a5e9af22439ac9e633707860073bd5d4c91e/src/Microsoft.Identity.Web/WebApiExtensions/MicrosoftIdentityWebApiAuthenticationBuilderExtensions.cs#L111)). These two functions return a new builder, using the same JwtBearerOptions and MicrosoftIdentityOptions used as arguments in the private function whose line I deleted. Those two returned builder objects either are or derive from the [MicrosoftIdentityWebApiAuthenticationBuilder](https://github.com/AzureAD/microsoft-identity-web/blob/48e4a5e9af22439ac9e633707860073bd5d4c91e/src/Microsoft.Identity.Web/WebApiExtensions/MicrosoftIdentityWebApiAuthenticationBuilder.cs#L45) where on line 45 in the constructor a Services.Configure call is made using those same JwtBearerOptions and MicrosoftIdentityOptions arguments.

Reply:
Thanks for the explanation @JoshLozensky. If you run the dev app which is webAppCallsWebApiCallsGraph, and add this to the service:

`services.Configure<MicrosoftIdentityOptions>(
                                           o => o.ClientSecret = "your_secret-here");`

and comment it out in the [appsettings.json](https://github.com/AzureAD/microsoft-identity-web/blob/master/tests/DevApps/WebAppCallsWebApiCallsGraph/TodoListService/appsettings.json#L7) then run them both as startup projects. I get this error:

An unhandled exception occurred while processing the request.

HttpRequestException: 500 InternalServerError MSAL.NetCore.4.55.0.0.MsalServiceException:
ErrorCode: invalid_client
Microsoft.Identity.Client.MsalServiceException: A configuration issue is preventing authentication - check the error message from the server for details. You can modify the configuration in the application registration portal. See https://aka.ms/msal-net-invalid-client for details. Original exception: AADSTS7000215: Invalid client secret provided. Ensure the secret being sent in the request is the client secret value, not the client secret ID, for a secret added to app 'a4c2469b-cf84-4145-8f5f-cb7bacf814bc'.

As it's not longer picking up the MicrosoftIdentityOptions. Does it repro like this for you?

Reply:
comment out the clientSecret in the appsettings.json. ^ not sure if that was clear.

Reply:
Actually, if I have this `services.Configure<MicrosoftIdentityOptions>(JwtBearerDefaults.AuthenticationScheme, o => o.ClientSecret = "your_secret-here");` it seems to work. 

@jmprieur thoughts on this requiring the authScheme now? 

@JoshLozensky were you able to repro the issue attached?

Reply:
Last thing, do we need to pass this in now? 
`Action<MicrosoftIdentityOptions> configureMicrosoftIdentityOptions,`

If not, please clean that up as part of this PR as well.

Reply:
Hey @jennyf19, I couldn't repro the exact same error in this branch, though I did get the same error when I ran it in the main repo branch. In this branch I did get a similar scenario with the call failing unless JwtBearerDefaults.AuthenticationScheme is added, but it doesn't appear to be related to the code in this PR.

It looks like in both services.Configure<MicrosoftIdentityOptions> scenarios a call to services.AddSingleton<IConfigureOptions>(name, configureOptions) is ultimately made (as seen in the image below). The difference is the name is set to either JwtBearerDefaults.AuthenticationScheme or Options.Options.DefaultName. It seems like the call without the authScheme would only work if the Options.Options.DefaultName was set to the same value as JwtBearerDefaults.AuthenticationScheme regardless of the changes made here.

![image](https://github.com/AzureAD/microsoft-identity-web/assets/103777376/c3fd4456-830b-4f33-98e0-21022e4f0b08)

Reply:
I cleaned up the Action<MicrosoftIdentityOptions> configureMicrosoftIdentityOptions, per your other comment.

Reply:
Thanks @JoshLozensky! It should fail if "similar scenario with the call failing unless JwtBearerDefaults.AuthenticationScheme is added".